larsim/EventGenerator/SingleGen_module.cc

```diff
@@ -725,7 +731,10 @@ namespace evgen {
 
     static TDatabasePDG pdgt;
     TParticlePDG* pdgp = pdgt.GetParticle(fPDG[i]);
-    if (pdgp) m = pdgp->Mass();
+    if (fPDG[i] == 1000020040)
+      m = kAlphaMass; // alpha particles
+    else if (pdgp)
+      m = pdgp->Mass();
```

Comment:
I suggest moving this functionality into a dedicated function in an anonymous namespace:

```c++
double particle_mass(int particle_id)
{
  constexpr double kAlphaMass = 3.727379240;
  if (particle_id == 1000020040)
    return kAlphaMass; // alpha particles

  static TDatabasePDG pdgt;
  TParticlePDG* pdgp = pdgt.GetParticle(particle_id);
  return pdgp ? pdgp->Mass() : 0.;
}
```

and then using it here and anywhere else it's needed in this file.